ivy/__init__.py

```diff
@@ -4,7 +4,19 @@
 from ivy._version import __version__ as __version__
 import builtins
 import numpy as np
-
+from . import globals
+
+# List of imported frameworks
+IMPORTED_FRAMEWORKS = [
+    m._name_ for m in globals().values() if getattr(m, "_package_", "") is not None
+]
+
+# Check if any imported module is not in the allowed list
+for framework_name in IMPORTED_FRAMEWORKS:
+    if framework_name.split(".")[0] != "numpy":
+        warnings.warn(
+            f"framework {framework_name} is not allowed. Only numpy are allowed."
+        )
 
```

Comment:
Hi @soma2000-lang, I've added a test file to this PR. Kindly pull the changes locally, and check for the import error. Your current solution causes a `circular import failure`, hence not allowing us to import ivy at all. Thanks 🙂